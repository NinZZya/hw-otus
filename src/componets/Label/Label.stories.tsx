import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Label, LabelProp } from "./Label";

export default {
  title: "Components/Label",
  component: Label,
} as Meta;

export const LabelWithText: React.FC<LabelProp> = () => <Label value="Label" />;
