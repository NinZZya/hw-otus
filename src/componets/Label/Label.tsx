import React from "react";
import styled from "@emotion/styled";

export interface LabelProp {
  value: string;
}

export const LabelWrapper = styled.span`
  display: flex;
  font-size: 3rem;
  padding: 1rem;
`;

export const Label: React.FC<LabelProp> = (props) => {
  const { value = "" } = props;

  return <LabelWrapper>{value}</LabelWrapper>;
};
