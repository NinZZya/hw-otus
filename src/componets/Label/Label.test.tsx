import React from "react";
import renderer from "react-test-renderer";
import { Label } from "./Label";

it("Label render coorectly", () => {
  const tree = renderer.create(<Label value="test" />).toJSON();

  expect(tree).toMatchSnapshot();
});
