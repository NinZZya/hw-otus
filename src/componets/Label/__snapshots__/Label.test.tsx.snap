// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label render coorectly 1`] = `
<span
  className="css-l3kone"
>
  test
</span>
`;
